src/registry/domain/repository.js

```diff
@@ -10,20 +10,22 @@ const ComponentsCache = require('./components-cache');
 const ComponentsDetails = require('./components-details');
 const packageInfo = require('../../../package.json');
 const registerTemplates = require('./register-templates');
-const S3 = require('./s3');
 const settings = require('../../resources/settings');
 const strings = require('../../resources');
 const validator = require('./validators');
 const versionHandler = require('./version-handler');
 
 module.exports = function(conf) {
-  const cdn = !conf.local && new S3(conf);
-  const repositorySource = conf.local ? 'local repository' : 's3 cdn';
+  const cdn = !conf.local && new conf.storage.adapter(conf.storage.options);
+  const options = !conf.local && conf.storage.options;
+  const repositorySource = conf.local
+    ? 'local repository'
+    : conf.storage.adapterType + ' cdn';
```

Comment:
Perhaps here we could have the adapter exposing his own name instead ?
maybe something like `conf.storage.adapter.type?`

Reply:
Funny I had it that way first..., but as stated before I wasn't a fan of 
adapter: require('oc/src/domain/registry/s3')

Reply:
I agree with you and I think we can avoid the user to set `adapter: require('oc/src/domain/registry/s3')` as that will be the default, so you'll only pass an adapter if you want to change that.